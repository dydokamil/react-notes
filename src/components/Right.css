.right .my-navbar {
  text-align: center;
}

button:disabled {
  cursor: default !important;
}

button:disabled:hover {
  color: gray;
  transform: none;
}
