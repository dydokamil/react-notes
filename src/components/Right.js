import React from 'react'
import { connect } from 'react-redux'
import consts from '../actions/consts'
import './Right.css'

export class Right extends React.Component {
  constructor (props) {
    super(props)
    this.state = { content: '', name: '', canSave: false }
  }

  componentWillReceiveProps = props => {
    if (
      props.notes &&
      props.notes.notes &&
      props.notes.selected &&
      props.notes.notes[props.notes.selected]
    ) {
      if ('content' in props.notes.notes[props.notes.selected]) {
        this.setState({
          content: props.notes.notes[props.notes.selected].content
        })
      } else {
        this.setState({ content: '' })
      }

      if ('name' in props.notes.notes[props.notes.selected]) {
        this.setState({ name: props.notes.notes[props.notes.selected].name })
      } else {
        this.setState({ name: '' })
      }
    }
    this.setState({ canSave: false })
  }

  onUpdateNoteWithParams = () => {
    this.props.onUpdateNote({
      content: this.state.content,
      name: this.state.name,
      _id: this.props.notes.notes[this.props.notes.selected]._id
    })
  }

  onChange = text => {
    this.setState({ content: text })
    this.setState({ canSave: true })
  }

  onChangeTitle = text => {
    this.setState({ name: text })
    this.setState({ canSave: true })
  }

  render () {
    return (
      <div className={this.props.className}>
        <div className="my-navbar">
          <button
            id="note-saver"
            disabled={!this.state.canSave}
            onClick={() => this.onUpdateNoteWithParams()}
          >
            <i className="fas fa-save" />
          </button>
        </div>
        {this.props.notes.selected && (
          <input
            id="title-input"
            className="title-editor"
            onChange={event => this.onChangeTitle(event.target.value)}
            placeholder="Title"
            value={this.state.name}
          />
        )}
        {this.props.notes.selected && (
          <textarea
            id="content-input"
            onChange={event => this.onChange(event.target.value)}
            placeholder="Type here..."
            value={this.state.content}
            className="editor"
          />
        )}
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    notes: state.notes
  }
}

const mapDispatchToProps = dispatch => ({
  onUpdateNote: body => {
    dispatch({ type: consts.UPDATE_NOTE_REQUEST, body })
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Right)
