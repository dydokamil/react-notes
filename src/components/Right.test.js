import React from 'react'
import { shallow } from 'enzyme'
import { Right } from './Right'

describe('Right component', () => {
  const notes = {}
  const props = { notes: { notes, selected: 1 } }
  const rightComponent = shallow(<Right {...props} />)
  it('should match the snapshot', () => {
    expect(rightComponent).toMatchSnapshot()
  })
  it('should render #note-saver button', () => {
    expect(rightComponent.find('#note-saver').exists()).toBe(true)
  })
  it('should render #title-input button', () => {
    expect(rightComponent.find('#title-input').exists()).toBe(true)
  })
  it('should render #content-input button', () => {
    expect(rightComponent.find('#content-input').exists()).toBe(true)
  })
  it('should call `onUpdateNoteWithParams` when #note-saver button is clicked', () => {
    // TODO
  })
})
