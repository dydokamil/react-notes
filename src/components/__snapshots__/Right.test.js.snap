// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Right component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Right
    notes={
        Object {
            "notes": Object {},
            "selected": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="my-navbar"
>
          <button
                    disabled={true}
                    id="note-saver"
                    onClick={[Function]}
          >
                    <i
                              className="fas fa-save"
                    />
          </button>
</div>,
        <input
          className="title-editor"
          id="title-input"
          onChange={[Function]}
          placeholder="Title"
          value=""
/>,
        <textarea
          className="editor"
          id="content-input"
          onChange={[Function]}
          placeholder="Type here..."
          value=""
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            disabled={true}
            id="note-saver"
            onClick={[Function]}
>
            <i
                        className="fas fa-save"
            />
</button>,
          "className": "my-navbar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-save"
/>,
            "disabled": true,
            "id": "note-saver",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-save",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "title-editor",
          "id": "title-input",
          "onChange": [Function],
          "placeholder": "Title",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "editor",
          "id": "content-input",
          "onChange": [Function],
          "placeholder": "Type here...",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="my-navbar"
>
            <button
                        disabled={true}
                        id="note-saver"
                        onClick={[Function]}
            >
                        <i
                                    className="fas fa-save"
                        />
            </button>
</div>,
          <input
            className="title-editor"
            id="title-input"
            onChange={[Function]}
            placeholder="Title"
            value=""
/>,
          <textarea
            className="editor"
            id="content-input"
            onChange={[Function]}
            placeholder="Type here..."
            value=""
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              disabled={true}
              id="note-saver"
              onClick={[Function]}
>
              <i
                            className="fas fa-save"
              />
</button>,
            "className": "my-navbar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-save"
/>,
              "disabled": true,
              "id": "note-saver",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-save",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "title-editor",
            "id": "title-input",
            "onChange": [Function],
            "placeholder": "Title",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "editor",
            "id": "content-input",
            "onChange": [Function],
            "placeholder": "Type here...",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
